app/src/main/java/com/example/bucketbuds/FriendsAdapter.java

```diff
@@ -0,0 +1,188 @@
+import com.parse.SaveCallback;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FriendsAdapter extends RecyclerView.Adapter<FriendsAdapter.ViewHolder> {
+
+    public static final String TAG = "FriendsAdapter";
+
+    Context context;
+    List<User> users;
+    Boolean friendsBool;
+    Fragment fragment;
+    List<String> friendsId;
+    UserPub currentUserPub;
+    public static final User currentUser = User.getCurrentUser();
+
+    // pass in context and list of users
+    public FriendsAdapter(Context context, List<User> users, UserPub currentUserPub, Boolean friendsBool, Fragment fragment, List<String> friendsId){
+        this.friendsBool = friendsBool;
+        this.context = context;
+        this.users = users;
+        this.currentUserPub = currentUserPub;
+        this.fragment = fragment;
+        this.friendsId = friendsId;
+    }
+
+    @NonNull
+    @NotNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
+        View view;
+        if (friendsBool) {
+            view = LayoutInflater.from(context).inflate(R.layout.item_friend, parent, false);
+        } else {
+            view = LayoutInflater.from(context).inflate(R.layout.item_add_friend, parent, false);
+        }
+        ViewHolder viewHolder = new ViewHolder(view);
+        return viewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull @NotNull ViewHolder holder, int position) {
+        User user = users.get(position);
+        holder.bind(user, friendsBool);
+    }
+
+    @Override
+    public int getItemCount() {
+        return users.size();
+    }
+
+    public class ViewHolder extends RecyclerView.ViewHolder {
+
+        ImageView ivProfileImage;
+        ImageView ivAddFriend;
+        ImageView ivRemoveFriend;
+        TextView tvUsername;
+        TextView tvFirstName;
+        TextView tvLastName;
+        TextView tvBucketCount;
+        TextView tvFriendCount;
+
+        public ViewHolder(View itemView) {
+            super(itemView);
+            if (friendsBool) {
+                ivProfileImage = itemView.findViewById(R.id.ivProfileImageIF);
+                tvUsername = itemView.findViewById(R.id.tvUsernameIF);
+                tvFirstName = itemView.findViewById(R.id.tvFirstNameIF);
+                tvLastName = itemView.findViewById(R.id.tvLastNameIF);
+                tvFriendCount = itemView.findViewById(R.id.tvFriendCountIF);
+                tvBucketCount = itemView.findViewById(R.id.tvBucketCountIF);
+                ivRemoveFriend = itemView.findViewById(R.id.ivRemoveFriendIF);
+            } else {
+                ivProfileImage = itemView.findViewById(R.id.ivProfileImageIAF);
+                tvUsername = itemView.findViewById(R.id.tvUsernameIAF);
+                tvFirstName = itemView.findViewById(R.id.tvFirstNameIAF);
+                tvLastName = itemView.findViewById(R.id.tvLastNameIAF);
```

Comment:
nit: Seems like these fields are present in both friends and not friends ... so you could just keep them out of the `if/else` block.
```
ivProfileImage =...
tvUsername = ...
...
if(friendsBool) {
  tvFriendCount = ...
} else {
  ivAddFriend=...
}
```

Reply:
they are present in both but they are corresponding to a different view IDs since they are on different layouts